packages/cli/src/run.ts

```diff
@@ -5,18 +5,15 @@ import * as logger from "./logger";
 import { ExitError } from "./errors";
 
 export async function runCmd(args: string[], cwd: string) {
-  let { packages, root } = await getPackages(cwd);
+  let { packages, rootDir } = await getPackages(cwd);
 
   const exactMatchingPackage = packages.find((pkg) => {
-    return (
-      pkg.packageJson.name === args[0] ||
-      path.relative(root.dir, pkg.dir) === args[0]
-    );
+    return pkg.packageJson.name === args[0] || pkg.relativeDir === args[0];
   });
 
   if (exactMatchingPackage) {
     const { code } = await spawn("yarn", args.slice(1), {
-      cwd: exactMatchingPackage.dir,
+      cwd: path.join(rootDir, exactMatchingPackage.relativeDir),
```

Comment:
most likely this can be simplified here too
```suggestion
      cwd: exactMatchingPackage.dir,
```